PSICover/PSICover.cs

```diff
@@ -163,31 +165,112 @@ class Analyzer {
+               code[block.SLine] = code[block.SLine].Insert (block.SCol, tag);
+            } else { // for multi line blocks w/wo other blocks on same line
+               string blockTag = $"<span class=\"tooltip covered\" data-title=\"{(hit ? hits[block.Id] + " hits" : "Code block not hit")}\"><span class=\"{(hit ? "hit" : "unhit")}\">";
+               string insideBlockTag = $"<span class=\"{(hit ? "hit" : "unhit")}\">";
+               code[block.ELine] = code[block.ELine].Insert (block.ECol, "</span></span>");
+               code[block.ELine] = code[block.ELine].Insert (GetFirstNonWhitespaceIndex (code[block.ELine]), insideBlockTag);
+               int lastLine = block.ELine - 1;
+               while (lastLine > block.SLine) {
+                  if (!string.IsNullOrEmpty (code[lastLine])) {
+                     int sIndex = GetFirstNonWhitespaceIndex (code[lastLine]);
+                     if (sIndex != code[lastLine].Length) {
+                        int lIndex = GetLastNonWhitespaceIndex (code[lastLine]);
+                        code[lastLine] = code[lastLine].Insert (lIndex + 1, "</span>");
+                        code[lastLine] = code[lastLine].Insert (sIndex, insideBlockTag);
+                     }
+                  }
+                  lastLine--;
+               }
+               code[block.SLine] = code[block.SLine].Insert (GetLastNonWhitespaceIndex (code[block.SLine]) + 1, "</span>");
+               code[block.SLine] = code[block.SLine].Insert (block.SCol, blockTag);
+            }
          }
          string htmlfile = $"{Dir}/HTML/{Path.GetFileNameWithoutExtension (file)}.html";
-
          string html = $$"""
-            <html><head><style>
-            .hit { background-color:aqua; }
-            .unhit { background-color:orange; }
-            </style></head>
-            <body><pre>
+            <html><head><title>{{Path.GetFileName(file)}}</title>
+            <link rel="stylesheet" type="text/css" href="{{Dir}}/styles/tooltip.css" />
+            </head><body><pre>
             {{string.Join ("\r\n", code)}}
             </pre></body></html>
             """;
          html = html.Replace ("\u00ab", "&lt;").Replace ("\u00bb", "&gt;");
          File.WriteAllText (htmlfile, html);
+         summary.Add (htmlfile, (Path.GetFullPath (file), hitCount, blocks.Count, Math.Round (100.0 * hitCount / blocks.Count, 1)));
       }
       int cBlocks = mBlocks.Count, cHit = hits.Count (a => a > 0);
       double percent = Math.Round (100.0 * cHit / cBlocks, 1);
-      Console.WriteLine ($"Coverage: {cHit}/{cBlocks}, {percent}%");
+      (int totalBlocks, int totalBlocksHit, double percentageHit) overall = (cBlocks, cHit, percent);
+      string summaryFilepath = $"{Dir}/HTML/Summary.html";
+      GenerateOutputSummary (summaryFilepath, summary, overall);
+      Process.Start (new ProcessStartInfo (summaryFilepath) { UseShellExecute = true });
+   }
+
+   int GetFirstNonWhitespaceIndex(string str) {

```

Comment:
Name seems to be too long. Try to minimize the function name. No space after function name.

Reply:
Done.